ironmon_tracker/CustomCode.lua

```diff
@@ -458,6 +467,50 @@ function CustomCode.logError(err)
 	end
 end
 
+---Properly executes each extension's startup function, respecting any extension dependencies (cascade loading)
+function CustomCode.executeExtensionStartups()
+	if not CustomCode.isEnabled() then return end
+
+	local startedExtensionKeys = {}
+
+	local _tryStartupExtensions = function(extensionsToStartup)
+		local anyNewStarted = false
+		local skippedExtensions = {}
+		for _, ext in pairs(extensionsToStartup or {}) do
+			local allRequiredAreStarted = true
+			for _, requiredKey in pairs(ext.selfObject and ext.selfObject.requiredExtKeys or {}) do
+				if not Utils.isNilOrEmpty(requiredKey) and not startedExtensionKeys[requiredKey] then
+					allRequiredAreStarted = false
+					break
+				end
+			end
```

Comment:
this checks for if an extension has any `requiredExtKeys` defined and if those required keys have already been started, but does not check if those required extensions *exist* to start-up

so say roguemon lists natdex as required, but the natdex extension has not been installed, this block will cause the later extension startup loop to cycle all 999 times as it will keep skipping roguemon due to natdex not being started (natdex cannot be started, it does not exist)

this might not be performance-costly, if all it's doing is skipping the extension, but I feel it should be handled better than "it loops 999 times, doing nothing each time" (and should inform the user so they are aware that the extension isn't starting because of a missing dependancy)

Reply:
in its current state, it wont loop 999 times if nothing new gets installed. each iteration/cascade check will try and install. if and only if a new thing gets installed, will it loop/iterate again to try and install more stuff. if at any point nothing changes from one iteration to the next, it simply exists